From e2363509bb88500eabc033f65c5c0c8ce7e692e8 Mon Sep 17 00:00:00 2001
Message-Id: <e2363509bb88500eabc033f65c5c0c8ce7e692e8.1659372189.git.thomas@gllm.fr>
In-Reply-To: <5b01e96eabd14521cfeb33b5ac11495b77f894cc.1659372189.git.thomas@gllm.fr>
References: <5b01e96eabd14521cfeb33b5ac11495b77f894cc.1659372189.git.thomas@gllm.fr>
From: Thomas Guillem <thomas@gllm.fr>
Date: Fri, 22 Jul 2022 09:16:08 +0200
Subject: [PATCH 08/19] vlc_fixups: don't declare swab() on Android NDK25

Fix build on Android NDK25, where swab() is a static inline function
defined in <unistd.h>.
---
 include/vlc_fixups.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/vlc_fixups.h b/include/vlc_fixups.h
index c03aa23cd0..88d9ebd762 100644
--- a/include/vlc_fixups.h
+++ b/include/vlc_fixups.h
@@ -375,8 +375,11 @@ extern "C" {
 #endif
 
 #ifndef HAVE_SWAB
+/* Android NDK25 have swab but configure fails to detect it */
+#ifndef __ANDROID__
 void swab (const void *, void *, ssize_t);
 #endif
+#endif
 
 /* Socket stuff */
 #ifndef HAVE_INET_PTON
-- 
2.35.1

